src/OrchardCore.Modules/OrchardCore.Tenants/Controllers/AdminController.cs

```diff
@@ -380,6 +380,20 @@ public async Task<IActionResult> Create(EditTenantViewModel model)
 
             await _shellHost.UpdateShellSettingsAsync(shellSettings);
 
+            if (action == "createAndSetup")
+            {
+                var dataProtector = _dataProtectorProvider.CreateProtector("Tokens").ToTimeLimitedDataProtector();
+
+                var entry = new ShellSettingsEntry
+                {
+                    Name = shellSettings.Name,
+                    ShellSettings = shellSettings,
+                    Token = dataProtector.Protect(shellSettings["Secret"], _clock.UtcNow.Add(new TimeSpan(24, 0, 0)))
+                };
+
+                return Redirect(HttpContext.GetEncodedUrl(entry));
```

Comment:
DRY this with the same logic in `Index`.

Reply:
The logic in the index checks the tenant state and the token value. Here we don't need to do that